osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/ClearMBusStatusOnAllChannelsCommandExecutorTest.java

```diff
@@ -0,0 +1,230 @@
+
+  private ClearMBusStatusOnAllChannelsCommandExecutor executor;
+
+  @BeforeEach
+  void setup() {
+    this.executor = new ClearMBusStatusOnAllChannelsCommandExecutor(this.dlmsObjectConfigService);
+    this.dlmsMessageListener = new LoggingDlmsMessageListener(null, null);
+  }
+
+  @Test
+  void testExecuteObjectNotFound() throws ProtocolAdapterException {
+    when(this.dlmsObjectConfigService.getAttributeAddress(any(), any(), any()))
+        .thenThrow(new ProtocolAdapterException("Object not found"));
+
+    final DlmsDevice dlmsDevice = new DlmsDevice();
+    dlmsDevice.setProtocol("SMR", "5.0.0");
+
+    assertThatExceptionOfType(ProtocolAdapterException.class)
+        .isThrownBy(
+            () -> {
+              this.executor.execute(
+                  this.connectionManager, dlmsDevice, this.dto, this.messageMetadata);
+            });
+  }
+
+  @Test
+  void shouldSkipIfStatusIsZero() throws ProtocolAdapterException, IOException {
+    final DlmsDevice dlmsDevice_5_1 = new DlmsDevice("SMR 5.1 device");
+    dlmsDevice_5_1.setProtocol("SMR", "5.1");
+
+    for (int channel = 1; channel <= 4; channel++) {
+      when(this.dlmsObjectConfigService.getAttributeAddress(
+              dlmsDevice_5_1, DlmsObjectType.READ_MBUS_STATUS, channel))
+          .thenReturn(
+              new AttributeAddress(
+                  InterfaceClass.EXTENDED_REGISTER.id(),
+                  OBIS_CODE_TEMPLATE_READ_STATUS.replaceAll("<c>", Integer.toString(channel)),
+                  ExtendedRegisterAttribute.VALUE.attributeId()));
+
+      when(this.dlmsObjectConfigService.getAttributeAddress(
+              dlmsDevice_5_1, DlmsObjectType.CLEAR_MBUS_STATUS, channel))
+          .thenReturn(
+              new AttributeAddress(
+                  InterfaceClass.DATA.id(),
+                  OBIS_CODE_TEMPLATE_CLEAR_STATUS.replaceAll("<c>", Integer.toString(channel)),
+                  DataAttribute.VALUE.attributeId()));
+
+      when(this.dlmsObjectConfigService.getAttributeAddress(
+              dlmsDevice_5_1, DlmsObjectType.CLIENT_SETUP_MBUS, channel))
+          .thenReturn(
+              new AttributeAddress(
+                  InterfaceClass.MBUS_CLIENT.id(),
+                  OBIS_CODE_TEMPLATE_MBUS_CLIENT_SETUP.replaceAll("<c>", Integer.toString(channel)),
+                  DataAttribute.VALUE.attributeId()));
+    }
+
+    when(this.connectionManager.getDlmsMessageListener()).thenReturn(this.dlmsMessageListener);
+    when(this.connectionManager.getConnection()).thenReturn(this.dlmsConnection);
+    when(this.dlmsConnection.get(this.attributeAddressArgumentCaptor.capture()))
+        .thenReturn(this.getResult);
+    when(this.getResult.getResultData()).thenReturn(DataObject.newUInteger32Data(0L));
+
+    this.executor.execute(this.connectionManager, dlmsDevice_5_1, this.dto, this.messageMetadata);
+
+    this.assertCurrentStatusAttributeAddresses(this.attributeAddressArgumentCaptor.getAllValues());
```

Comment:
Would it make sense to verify here as well that the clear status and reset alarm methods are not called?

Reply:
added asserts for these methods